### PR TITLE
Add resiliency in accessing EKS cluster after creating access entry for it

### LINKED ISSUE #37844: Unable to enroll EKS cluster via Discover
Expected behavior:

Following the EKS Discover wizard results in a successfully enrolled cluster

Current behavior:

The cluster is found after setting up the integration, but after selecting the cluster and clicking "ENROLL EKS CLUSTER" the following error prompt is received.

image

### PR DESCRIPTION
**Pull request by AntonAM**
There is race condition when accessing EKS cluster right after creating access entry/associating policy for it - permissions sometimes don't have enough time to propagate to the cluster and it fails auth. In repro I was able to just repeatedly click "retry" button and see it sometimes to fail and sometimes to succeed. If you manually add access entry (so it doesn't have to add entry itself) it would always succeed. This PR adds a backoff loop to retry our attempts to access cluster to handle this race condition.

Fixes #37844

### CODE REVIEW

File: lib/integrations/awsoidc/eks_enroll_clusters.go [hunk 1]
```diff
@@ -467,14 +468,25 @@ func getHelmSettings() (*helmCli.EnvSettings, error) {
...
-
-	releases, err := listCmd.Run()
+	var releases []*release.Release
+	var err error
+	// We setup a little backoff loop because sometimes access entry auth needs a bit more time to propagate and take
+	// effect, so we could get errors when trying to access cluster right after giving us permissions to do so.
+	for attempt := 1; attempt <= 3; attempt++ {
+		listCmd := action.NewList(actionConfig)
+		releases, err = listCmd.Run()
+		if err != nil {
```
Comment (bl-nero) [inline lib/integrations/awsoidc/eks_enroll_clusters.go:478]: Is there a way to tell if the error actually stems from the permission propagation issue and only loop in this case?

Comment (AntonAM) [reply lib/integrations/awsoidc/eks_enroll_clusters.go]: I saw at least two different type of errors that can occur and I'm not sure if some other can occur as well, so I think it's better to retry just on any error.

File: lib/integrations/awsoidc/eks_enroll_clusters.go [hunk 2]
```diff
@@ -467,14 +468,25 @@ func getHelmSettings() (*helmCli.EnvSettings, error) {
...
 // checkAgentAlreadyInstalled checks through the Helm if teleport-kube-agent chart was already installed in the EKS cluster.
 func checkAgentAlreadyInstalled(actionConfig *action.Configuration) (bool, error) {
-	listCmd := action.NewList(actionConfig)
-
-	releases, err := listCmd.Run()
+	var releases []*release.Release
+	var err error
+	// We setup a little backoff loop because sometimes access entry auth needs a bit more time to propagate and take
+	// effect, so we could get errors when trying to access cluster right after giving us permissions to do so.
+	for attempt := 1; attempt <= 3; attempt++ {
```
[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/integrations/awsoidc/eks_enroll_clusters.go]: I did, but it seemed like too heavy a gun for what we tried to achieve here. 
I've changed to use retry utils, but I'm not sure if it's better, there's no good way with retry utils to go for a specific attempts number, now it's about timing, but I can leave it like this if you think it's better, I don't have strong preference.

[the remark this answers is not in the document]
Comment (AntonAM) [reply lib/integrations/awsoidc/eks_enroll_clusters.go:475]: Reverted usage of `retry` as discussed.

**Review by rosstimothy — APPROVED** [review]

**Review by bl-nero — APPROVED** [review]

**Pull request merged**